scripts/zones/RoMaeve/Zone.lua

```diff
@@ -35,21 +35,36 @@ function onRegionEnter(player, region)
 end
 
 function onGameHour(zone)
-    local hour = VanadielHour()
+    -- Make Ro'Maeve come to life between 6pm and 6am during a full moon
+    local romaeveActive = moongate1:getLocalVar("romaeveActive")
```

Comment:
moongate1 hasn't been defined yet~!

Reply:
woopsie

Reply:
my fault

Reply:
```lua
function onGameHour(zone)
    local hour = VanadielHour()

    -- Make Ro'Maeve come to life between 6pm and 6am during a full moon
    if IsMoonFull() and (hour >= 18 or hour < 6) then
        local moongate1 = GetNPCByID(ID.npc.MOONGATE_OFFSET)
        local moongate2 = GetNPCByID(ID.npc.MOONGATE_OFFSET + 1)
        if moongate1:getLocalVar("romaeveActive") == 0 then

            -- Loop over the affected NPCs: Moongates, bridges and fountain
            for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET + 7 do
                GetNPCByID(i):setAnimation(tpz.anim.OPEN_DOOR) -- Open them
            end
            moongate2:untargetable(true)
            moongate1:untargetable(true)
            moongate1:setLocalVar("romaeveActive", 1) -- Make this loop unavailable after firing
        end

    -- Clean up at 6am
    elseif hour == 6 then
        local moongate1 = GetNPCByID(ID.npc.MOONGATE_OFFSET)
        local moongate2 = GetNPCByID(ID.npc.MOONGATE_OFFSET + 1)
        if moongate1:getLocalVar("romaeveActive") == 1 then
            for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET + 7 do
                GetNPCByID(i):setAnimation(tpz.anim.CLOSE_DOOR)
            end
            moongate2:untargetable(false)
            moongate1:untargetable(false)
            moongate1:setLocalVar("romaeveActive", 0) -- Make loop available again
        end
    end
end
```
fixed version of what I wrote before

Reply:
Sorry, was sleeping myself there. Changed as proposed, kept `vanadielHour` though.